lib/modules/manager/asdf/extract.ts

```diff
@@ -0,0 +1,68 @@
+import is from '@sindresorhus/is';
+import { logger } from '../../../logger';
+import { isSkipComment } from '../../../util/ignore';
+import { regEx } from '../../../util/regex';
+import { GithubTagsDatasource } from '../../datasource/github-tags';
+import * as nodeVersioning from '../../versioning/node';
+import type { PackageDependency, PackageFile } from '../types';
+
+const upgradeableTooling: Record<
+  string,
+  Pick<
+    PackageDependency,
+    'depName' | 'datasource' | 'packageName' | 'versioning'
+  >
+> = {
+  nodejs: {
+    depName: 'node',
+    datasource: GithubTagsDatasource.id,
+    packageName: 'nodejs/node',
+    versioning: nodeVersioning.id,
+  },
+};
+
+export function extractPackageFile(content: string): PackageFile | null {
+  logger.trace('asdf.extractPackageFile()');
+
+  const regex = regEx(
+    /^(?<toolName>(\w+))\s+((?<version>(\d[\d.]+\d))|(?<altVersion>([^\s]+)))([^#]*(#(?<comment>(.*))))?/gm
```

Comment:
i think we should use a simpler regex for version. renovate will handle invalid versions later automatically.

```suggestion
    /^(?<toolName>\w+)\s+(?<version>[^\s]+)\s*(?:#(?<comment>.*))?/gm
```

can there be arbitrary chars before the comment?

Reply:
I wanted to differentiate between versions that were supported and those that were not

using one field means I then have to test it again later in the code - happy to change if you insist but just letting you know there was reason behind this madness

Reply:
> can there be arbitrary chars before the comment?

there could be other version fallbacks - I am choosing to only maintain the first version if it is a supported version string i.e. 1.2.3

Reply:
You don't need to test the version in manager, renovate will use the versioning to validate the version later anyways and add an appropriate `skipReason`

Ok, so ignore the fallback versions for now and document that in readme.
```suggestion
     /^(?<toolName>\w+)\s+(?<version>[^\s#]+)(?:\s+[^\s#]+)*(?:\s+#(?<comment>.*))?/gm
```

https://asdf-vm.com/manage/configuration.html#tool-versions